sec_certs_page/dashboard/data.py

```diff
@@ -0,0 +1,399 @@
+            data = list(cursor)
+
+            if not data:
+                logger.warning("FIPS query returned no results")
+                return pd.DataFrame()
+
+            df = pd.DataFrame(data)
+            logger.info(f"Returned {len(df)} FIPS records")
+            return df
+
+        except Exception as e:
+            logger.exception("Error fetching FIPS data from MongoDB")
+            raise e
+
+    def get_dataframe(
+        self,
+        collection_type: CollectionName,
+        filter_values: dict[str, Any] | None = None,
+    ) -> pd.DataFrame:
+        """Get dataset from MongoDB based on collection type.
+
+        :param collection_type: The collection to query (CC or FIPS)
+        :param filter_values: Optional dictionary mapping filter IDs to values
+        :return: Dataset as DataFrame
+        :raises ValueError: If collection_type is not supported
+        """
+        if collection_type == CollectionName.CommonCriteria:
+            return self.get_cc_dataframe(filter_values)
+        elif collection_type == CollectionName.FIPS140:
+            return self.get_fips_dataframe(filter_values)
+        else:
+            raise ValueError(f"Unsupported collection type: {collection_type}")
+
+    def execute_aggregation_pipeline(
+        self,
+        collection_type: CollectionName,
+        pipeline: list[dict[str, Any]],
+    ) -> pd.DataFrame:
+        """Execute a MongoDB aggregation pipeline and return results as DataFrame.
+
+        This is used for charts with custom aggregation queries built from
+        user-defined axis configurations and filters.
+
+        :param collection_type: The collection to query (CC or FIPS)
+        :param pipeline: MongoDB aggregation pipeline stages
+        :return: Aggregated data as DataFrame
+        """
+        collection_map = {
+            CollectionName.CommonCriteria: self.mongo.db.cc,  # pyright: ignore[reportOptionalMemberAccess]
+            CollectionName.FIPS140: self.mongo.db.fips,  # pyright: ignore[reportOptionalMemberAccess]
+        }
+
+        collection = collection_map.get(collection_type)
+        if collection is None:
+            raise ValueError(f"Unsupported collection type: {collection_type}")
+
+        logger.info(f"Executing aggregation pipeline on {collection_type.value}: {pipeline}")
+
+        try:
+            cursor = collection.aggregate(pipeline)
+            data = list(cursor)
+
+            if not data:
+                logger.warning(f"Aggregation pipeline returned no results for {collection_type.value}")
+                return pd.DataFrame()
+
+            df = pd.DataFrame(data)
+            logger.info(f"Aggregation returned {len(df)} records")
+            return df
+
+        except Exception as e:
+            logger.exception(f"Error executing aggregation pipeline on {collection_type.value}")
+            raise e
+
+    def get_distinct_values(self, field: str, dataset_type) -> list[Any]:
```

Comment:
Why does this have a string `dataset_type` while other similar usecases have `CollectionName`?